File: benchmark/fast/test_primos.cc
#include <iostream>
using namespace std;

bool ehPrimo(int n) {
    if (n < 2) return false;
    if (n == 2) return true;
    if (n % 2 == 0) return false;
    int limite = n / 2;
    for (int i = 3; i <= limite; i++) {
        if (n % i == 0) return false;
    }
    return true;
}

int contarPrimos(int limite) {
    int contador = 0;
    for (int numero = 2; numero <= limite; numero++) {
        if (ehPrimo(numero)) contador++;
    }
    return contador;
}

int main() {
    int limite = 1000000;
    int resultado = contarPrimos(limite);
    cout << "RESULT:" << resultado << endl;
    return 0;
}

File: benchmark/fast/test_primos.py
def eh_primo(n):
    if n < 2:
        return False
    if n == 2:
        return True
    if n % 2 == 0:
        return False
    limite = n // 2
    for i in range(3, limite + 1):
        if n % i == 0:
            return False
    return True

def contar_primos(limite):
    contador = 0
    for numero in range(2, limite + 1):
        if eh_primo(numero):
            contador += 1
    return contador

limite = 1_000_000
resultado = contar_primos(limite)
print("RESULT:" + str(resultado))
